modules/utils/include/hephaestus/utils/string/string_utils.h

```diff
@@ -19,9 +19,23 @@ namespace heph::utils {
 /// constexpr auto truncated = truncate(str, start_token, end_token, false);
 /// std::cout << truncated << '\n'; // $ to/some/file
 /// @endcode
-constexpr auto truncate(std::string_view str, std::string_view start_token,
-                        std::string_view end_token = std::string_view(""),
-                        bool include_end_token = true) -> std::string_view {
+[[nodiscard]] constexpr auto truncate(std::string_view str, std::string_view start_token,
+                                      std::string_view end_token = std::string_view(""),
+                                      bool include_end_token = true) -> std::string_view;
+
+/// aNy_CaSe -> ANY_CASE
+[[nodiscard]] auto toUpperCase(const std::string_view& any_case) -> std::string;
+
+/// camelCase -> camel_case
+[[nodiscard]] auto toSnakeCase(const std::string_view& camel_case) -> std::string;
+
+/// camelCase -> CAMEL_CASE
+[[nodiscard]] auto toScreamingSnakeCase(const std::string_view& camel_case) -> std::string;
+
+/* --- Implementation --- */
```

Comment:
nit. I would be more verbose here to split, but prob we should be consistent across the code base.
```suggestion
// -------------------------------------------- 
// --- Implementation ------------------------
// -------------------------------------------- 
```

Reply:
Have another look. Wdyt?

On the white space after the comments, we are not consistent at the moment. Would you want it or not?

Reply:
ah no that was a typo there shouldn't be any trailing whitespace and you can set your editor to remove themn

Reply:
ok, sounds good